rcl_lifecycle/src/default_state_machine.c

```diff
@@ -695,14 +699,27 @@ rcl_lifecycle_init_default_state_machine(
 fail:
   // If rcl_lifecycle_transition_map_fini() fails, it will clobber the error string here.
   // Concatenate the error strings if that happens
+  default_allocator = rcl_get_default_allocator();
+
   if (rcl_error_is_set()) {
-    fail_error_message = rcl_get_error_string().str;
+    init_error_string = rcutils_strdup(rcl_get_error_string().str, default_allocator);
+    if (init_error_string != NULL) {
+      fail_error_message = init_error_string;
+    } else {
```

Comment:
I don't think this is necessary. If `init_error_string` is null, then so be it. Only set the `fini_error_message` to the `strdup` failure.

Reply:
> If init_error_string is null, then so be it. Only set the fini_error_message to the strdup failure.  

Sorry. I don't understand your comments.  
Your comments describe what to do for else code.  

```c++
    if (init_error_string != NULL) {
      fail_error_message = init_error_string;
    } else {
      fail_error_message = "Error while duplicating strings by rcutils_strdup() !";
    }
```